### DISCUSSION TITLE
Failure on Update for Microsoft Defender Antivirus antimalware platform - KB4052623

### DISCUSSION THREAD

**Discussion by Mikebeem** [actions/virtual-environments#6231] [General questions] [answered]
### Description

The image build for Windows 2022 is failing for some while now in our environment. It fails on Update for Microsoft Defender Antivirus antimalware platform - KB4052623:

==> vhd: Provisioning with powershell script: /agent/_work/1/s/images/win/scripts/Installers/Wait-WindowsUpdatesForInstall.ps1
    vhd: There is an error encountered:
    vhd:  InProgress: Update for Microsoft Defender Antivirus antimalware platform - KB4052623 (Version 4.18.2207.7)
    vhd: Waiting 120 seconds before retrying. Retries left: 9
    vhd: There is an error encountered:
    vhd:  InProgress: Update for Microsoft Defender Antivirus antimalware platform - KB4052623 (Version 4.18.2207.7)
    vhd: Waiting 120 seconds before retrying. Retries left: 8
    vhd: There is an error encountered:
    vhd:  InProgress: Update for Microsoft Defender Antivirus antimalware platform - KB4052623 (Version 4.18.2207.7)
    vhd: Waiting 120 seconds before retrying. Retries left: 7
    vhd: There is an error encountered:
    vhd:  InProgress: Update for Microsoft Defender Antivirus antimalware platform - KB4052623 (Version 4.18.2207.7)
    vhd: Waiting 120 seconds before retrying. Retries left: 6
    vhd: There is an error encountered:
    vhd:  InProgress: Update for Microsoft Defender Antivirus antimalware platform - KB4052623 (Version 4.18.2207.7)
    vhd: Waiting 120 seconds before retrying. Retries left: 5
    vhd: There is an error encountered:
    vhd:  InProgress: Update for Microsoft Defender Antivirus antimalware platform - KB4052623 (Version 4.18.2207.7)
    vhd: Waiting 120 seconds before retrying. Retries left: 4
    vhd: There is an error encountered:
    vhd:  InProgress: Update for Microsoft Defender Antivirus antimalware platform - KB4052623 (Version 4.18.2207.7)
    vhd: Waiting 120 seconds before retrying. Retries left: 3
    vhd: There is an error encountered:
    vhd:  InProgress: Update for Microsoft Defender Antivirus antimalware platform - KB4052623 (Version 4.18.2207.7)
    vhd: Waiting 120 seconds before retrying. Retries left: 2
    vhd: There is an error encountered:
    vhd:  InProgress: Update for Microsoft Defender Antivirus antimalware platform - KB4052623 (Version 4.18.2207.7)
    vhd: Waiting 120 seconds before retrying. Retries left: 1
    vhd: There is an error encountered:
    vhd:  InProgress: Update for Microsoft Defender Antivirus antimalware platform - KB4052623 (Version 4.18.2207.7)
==> vhd: Provisioning step had errors: Running the cleanup provisioner, if present...

### Platforms affected

- [X] Azure DevOps
- [ ] GitHub Actions

### Runner images affected

- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 10.15
- [ ] macOS 11
- [ ] macOS 12
- [ ] Windows Server 2019
- [X] Windows Server 2022

### Image version and build link

Running build on an ubuntu1804 server.

### Is it regression?

Image Version: 20220828.1

### Expected behavior

Is installs updates successful.

### Actual behavior

It fails every day.

### Repro steps

Run the pipeline. I'm using location West Europe. 

I don't understand why we are getting this issue, as Microsoft's pipelines don't show this error.

It looks like Install-WindowsUpdates.ps1 is ignoring [Microsoft Defender Antivirus](https://github.com/actions/runner-images/blob/6616fca6147fc1a02483ca6967f61ab233847dba/images/win/scripts/Installers/Install-WindowsUpdates.ps1#L26), but after a restart Wait-WindowsUpdatesForInstall.ps1 fails on it.

**Comment by iedgeleybistech** [comment]
Funny you just posted this - I was having this issue earlier in the week, and out of curiosity I extended the Wait-WindowsUpdatesForInstall window to a few hours and this update never seemed to complete. I just reran my pipeline today and it built the image with no issues, pulling from the latest 2022 definition here. Came looking for an issue I could reference for our records...

**Reply by Mikebeem** [reply]
For us it was gone for one day, but now it still fails on this update every day.

**Reply by al-cheb** [reply]
In scope of this https://github.com/actions/runner-images/pull/6259 PR we excluded 'Windows Defender' check.

**Comment by Mikebeem** [comment, marked as answer]
> In scope of this #6259 PR we excluded 'Windows Defender' check.

Thanks @al-cheb!
Build is working now.

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
